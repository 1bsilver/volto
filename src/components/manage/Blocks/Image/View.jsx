/**
 * View image block.
 * @module components/manage/Blocks/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import ImageFromUrl from '@plone/volto/components/theme/ImageFromUrl/ImageFromUrl';

import { flattenToAppURL, isInternalURL } from '@plone/volto/helpers';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'block image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <ImageFromUrl
              url={data.url}
              alt={data.alt}
              className={cx({
                'full-width': data.align === 'full',
                large: data.size === 'l',
                medium: data.size === 'm',
                small: data.size === 's',
              })}
              size={(() => {
                switch (data.size) {
                  case 'm':
                    return 'preview';

                  case 's':
                    return 'mini';

                  case 'l':
                  default:
                    return null;
                }
              })()}
            />
          );
          if (data.href) {
            if (!isInternalURL(data.href)) {
              return (
                <a
                  target={data.openLinkInNewTab ? '_blank' : null}
                  href={data.href}
                >
                  {image}
                </a>
              );
            } else {
              return (
                <Link
                  to={flattenToAppURL(data.href)}
                  target={data.openLinkInNewTab ? '_blank' : null}
                >
                  {image}
                </Link>
              );
            }
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
