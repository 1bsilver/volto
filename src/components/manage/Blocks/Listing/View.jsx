import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import ListingBody from '@plone/volto/components/manage/Blocks/Listing/ListingBody';

const View = ({ data, properties, block, path }) => {
  return (
    <div className={cx('block listing', data.template)}>
      <ListingBody
        data={data}
        properties={properties}
        block={block}
        path={path}
      />
    </div>
  );
};

View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  properties: PropTypes.objectOf(PropTypes.any).isRequired,
  block: PropTypes.string,
};

export default View;
