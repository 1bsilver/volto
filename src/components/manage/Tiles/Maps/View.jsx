/**
 * View map tile.
 * @module components/manage/Tiles/Maps/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import './styles.css';

/**
 * View image tile class.
 * @class View
 * @extends Component
 */
const View = ({ data }) => (
  <p
    className={['tile', 'image', 'align', data.align]
      .filter(e => !!e)
      .join(' ')}
  >
    <Container>
      <iframe
        title="Embeded Google Maps"
        src={data.url}
        className="google-map"
        frameborder="0"
        allowfullscreen
      />
    </Container>
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
