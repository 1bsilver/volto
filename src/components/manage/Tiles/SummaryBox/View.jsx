/**
 * View summary box tile.
 * @module components/manage/Tiles/SummaryBox/View
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { get } from 'lodash';
import { Card, Image } from 'semantic-ui-react';

import {
  getSummaryBoxContent,
  resetSummaryBoxContent,
} from '../../../../actions';

const parseContent = content => ({
  url: content['@id'],
  image: get(content, 'image.scales.mini.download', undefined),
  title: content.title,
  description: content.description,
});

/**
 * Summary box view component.
 * @class View
 * @extends Component
 */
export class View extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    getSummaryBoxContent: PropTypes.func.isRequired,
    resetSummaryBoxContent: PropTypes.func.isRequired,
    content: PropTypes.objectOf(PropTypes.any),
  };

  static defaultProps = {
    content: {},
  };

  /**
   * Constructor
   * @method constructor
   * @param {Object} props Component properties
   * @constructs SummaryBoxEditor
   */
  constructor(props) {
    super(props);

    this.state = parseContent(props.content);
  }

  /**
   * Component did mount
   * @method componentDidMount
   * @returns {undefined}
   */
  componentDidMount() {
    const { selectedItem } = this.props.data;
    if (selectedItem) {
      this.props.getSummaryBoxContent(selectedItem);
    }
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Props that will be received
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (Object.keys(nextProps.content).length) {
      this.setState(parseContent(nextProps.content));
    }
  }

  /**
   * Component will unmount. Reset loaded content.
   * @method componentWillUnmount
   * @returns {undefined}
   */
  componentWillUnmount() {
    this.props.resetSummaryBoxContent();
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const { url, image, title, description } = this.state;

    return (
      url && (
        <Card floated="left">
          {image && <Image clearing src={image} alt={title} />}
          <Card.Content>
            <Card.Header>{title}</Card.Header>
            <Card.Description>{description}</Card.Description>
          </Card.Content>
        </Card>
      )
    );
  }
}

export default connect(
  state => ({
    content: state.summaryBox.content,
  }),
  dispatch =>
    bindActionCreators(
      {
        getSummaryBoxContent,
        resetSummaryBoxContent,
      },
      dispatch,
    ),
)(View);
