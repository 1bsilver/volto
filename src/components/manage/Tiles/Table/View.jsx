/**
 * View table tile.
 * @module components/manage/Tiles/Table/View
 */

import React from 'react';
import PropTypes from 'prop-types';

/**
 * View table tile class.
 * @class View
 * @extends Component
 */
const View = ({ props }) => <span>table</span>;

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  props: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
