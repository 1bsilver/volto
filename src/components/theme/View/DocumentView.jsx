/**
 * Document view component.
 * @module components/theme/View/DocumentView
 */

import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

/**
 * Component to display the document view.
 * @function DocumentView
 * @param {Object} content Content object.
 * @returns {string} Markup of the component.
 */
const DocumentView = ({ content }) => (
  <div id="page-home">
    <Helmet title={content.title} />
    <h1>{content.title}</h1>
    {content.description && (
      <p className="description">{content.description}</p>
    )}
    {content.text && (
      <p dangerouslySetInnerHTML={{ __html: content.text.data }} />
    )}
  </div>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
DocumentView.propTypes = {
  /**
   * Content of the object
   */
  content: PropTypes.shape({
    /**
     * Title of the object
     */
    title: PropTypes.string,
    /**
     * Description of the object
     */
    description: PropTypes.string,
    /**
     * Text of the object
     */
    text: PropTypes.shape({
      /**
       * Data of the text of the object
       */
      data: PropTypes.string,
    }),
  }).isRequired,
};

export default DocumentView;
