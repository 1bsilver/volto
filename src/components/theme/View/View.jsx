/**
 * View container.
 * @module components/theme/View/View
 */

import React, { PropTypes, Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { getContent } from '../../../actions';
import { SummaryView, DocumentView } from '../../../components';

@connect(
  state => ({
    content: state.content.content,
    pathname: state.routing.locationBeforeTransitions.pathname,
  }),
  dispatch => bindActionCreators({ getContent }, dispatch),
)
/**
 * View container class.
 * @class View
 * @extends Component
 */
export default class View extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    getContent: PropTypes.func.isRequired,
    pathname: PropTypes.string.isRequired,
    content: PropTypes.shape({
      '@type': PropTypes.string,
    }),
  }

  /**
   * Default properties.
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    content: null,
  }

  /**
   * Component will mount
   * @method componentWillMount
   * @returns {undefined}
   */
  componentWillMount() {
    this.props.getContent(this.props.pathname);
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.pathname !== this.props.pathname) {
      this.props.getContent(nextProps.pathname);
    }
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    if (!this.props.content) {
      return <span />;
    }

    switch (this.props.content['@type']) {
      case 'Folder':
        return <SummaryView content={this.props.content} />;
      case 'Document':
        return <DocumentView content={this.props.content} />;
      default:
        return <span />;
    }
  }
}
