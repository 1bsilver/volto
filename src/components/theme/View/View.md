View example source:

```jsx_ noeditor
import {StaticRouter} from 'react-router-dom';
import View from './View';

const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;

const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
  users: {
    reset: {
      loading: '',
    },
  },
  apierror: {},
  actions: {},
  content: { get: {} },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <View
        listActions={() => {}}
        pathname=""
        location={{ search: '/hello', pathname: '/' }}
        getContent={() => {}}
        versionId=""
        content={{
          layout: 'PLone',
          allow_discussion: true,
          title: 'plone',
          description: 'plone CMS',
          '@type': '@Document',
          is_folderish: true,
          subjects: [],
        }}
      />
    </StaticRouter>
  </Provider>
</div>;
```
