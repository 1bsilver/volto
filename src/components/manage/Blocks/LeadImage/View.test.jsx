import React from 'react';
import renderer from 'react-test-renderer';
import View from './View';

test('renders a view image component', () => {
  const component = renderer.create(
    <View
      data={{}}
      properties={{
        image: {
          'content-type': 'image/jpeg',
          download:
            'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
          filename: 'image.jpg',
          height: 1280,
          scales: {
            icon: {
              download:
                'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
              height: 21,
              width: 32,
            },
            large: {
              download:
                'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
              height: 512,
              width: 768,
            },
            listing: {
              download:
                'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
              height: 10,
              width: 16,
            },
            mini: {
              download:
                'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
              height: 133,
              width: 200,
            },
            preview: {
              download:
                'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
              height: 266,
              width: 400,
            },
            thumb: {
              download:
                'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
              height: 85,
              width: 128,
            },
            tile: {
              download:
                'http://localhost:8080/Plone/test-images/@@images/image.jpeg',
              height: 42,
              width: 64,
            },
          },
          size: 186918,
          width: 1920,
        },
      }}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
