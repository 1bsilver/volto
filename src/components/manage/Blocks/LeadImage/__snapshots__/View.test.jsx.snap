// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a view image component 1`] = `
<p
  className="block image align center"
>
  <img
    alt=""
    className=""
    loading="lazy"
    role="img"
    sizes={null}
    src="/test-images/@@images/image.jpeg"
    srcSet="/test-images/@@images/image.jpeg 768w, /test-images/@@images/image.jpeg 400w, /test-images/@@images/image.jpeg 200w, /test-images/@@images/image.jpeg 128w, /test-images/@@images/image.jpeg 64w, /test-images/@@images/image.jpeg 32w, /test-images/@@images/image.jpeg 16w"
  />
</p>
`;
